### PR TITLE
Hierarchical Charts Popurri

### PR DESCRIPTION
**Pull request by BennuFire**
It contains some commits of your own. Check if merge w/o conflict

Hierarchical  Charts Added
Pie Chart Animation Options added
Pie Chart Percentage Default true

### CODE REVIEW

**Comment by BennuFire** [comment]
image

File: src/chart/CirclePackagingChart.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { ChartProps } from './Chart';
+import CirclePackagingVisualization from './visualizations/CirclePackagingVisualization';
+
+
+/**
+ * Embeds a CirclePackagingChart (from Charts) into NeoDash.
+ */
+const NeoCirclePackagingChart = (props: ChartProps) => {
```
Comment (nielsdejong) [inline src/chart/CirclePackagingChart.tsx:9]: Shouldn't this be 'Circle Packing'?

Comment (BennuFire) [reply src/chart/CirclePackagingChart.tsx:9]: U r right! There r a couple modifications to here then in order to adjust the naming.

File: src/chart/visualizations/SunburstVisualization.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                cornerRadius={cornerRadius}
+                margin={{
+                    top: marginTop,
+                    right: marginRight,
+                    bottom: (legend) ? legendHeight + marginBottom : marginBottom,
+                    left: marginLeft
+                }}
+                animate={true}
+                arcLabelsSkipAngle={arcLabelsSkipAngle}
+                colors={{scheme: 'nivo'}}/></>)
```
[the remark this answers is not in the document]
Comment (BennuFire) [reply src/chart/visualizations/SunburstVisualization.tsx:68]: Yep

File: src/chart/visualizations/Utils.ts [hunk 3]
```diff
@@ -35,4 +35,390 @@ export function checkResultKeys(first: Neo4jRecord, keys: string[]) {
     }
 
     return false
-}
\ No newline at end of file
+}
+
+export const search = (tree, value, key = 'id', reverse = false) => {
```
[the remark this answers is not in the document]
Comment (BennuFire) [reply src/chart/visualizations/Utils.ts:40]: Here I'm trying to look for the specific node on the tree structure based on the key property that must have a given value.

File: src/chart/visualizations/Utils.ts [hunk 4]
```diff
@@ -35,4 +35,390 @@ export function checkResultKeys(first: Neo4jRecord, keys: string[]) {
...
+
+                holder = search(holder, val, 'name');
+            }
+        }
+        holder.loc = value;
+        return data
+    }, []);
+}
+
+export const dataTestHierarcy =  {
```
[the remark this answers is not in the document]
Comment (BennuFire) [reply src/chart/visualizations/Utils.ts:101]: Nope. I was planning to keep it somewhere considering is the test data they use on their 'no problems' drilldown example, So I was able to track the bug on big trees problem. We can delete it eventually.

File: src/config/ReportConfig.tsx [hunk 5]
```diff
@@ -389,6 +397,258 @@ export const REPORT_TYPES = {
                 type: SELECTION_TYPES.NUMBER,
                 default: 0
             },
+            "activeOuterRadiusOffset": {
```
[the remark this answers is not in the document]
Comment (BennuFire) [reply src/config/ReportConfig.tsx:400]: https://www.youtube.com/watch?v=t6-cCh9bTG4&ab_channel=SyrebralVibes

File: src/config/ReportConfig.tsx [hunk 6]
```diff
@@ -389,6 +397,258 @@ export const REPORT_TYPES = {
...
+                type: SELECTION_TYPES.LIST,
+                values: [true, false],
+                default: true
+            }
+        }
+    },
+    "sunburst": {
+        label: "Sunburst Chart",
+        component: SunburstChart,
+        helperText: <div>A pie chart with drill down expects two fields: a <code>PATH</code> and a <code>VALUE</code>.</div>,
```
[the remark this answers is not in the document]
Comment (BennuFire) [reply src/config/ReportConfig.tsx:451]: Copy Paste Demons

File: src/report/ReportQueryRunner.ts [hunk 7]
```diff
@@ -72,6 +72,7 @@ export async function runCypherQuery(driver,
             query = "CALL { " + query + "} RETURN * LIMIT " + (rowLimit + 1)
         }
     }
+    console.log(query);
```
[the remark this answers is not in the document]
Comment (BennuFire) [reply src/report/ReportQueryRunner.ts:75]: Sure

File: src/config/ReportConfig.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (BennuFire) [reply src/config/ReportConfig.tsx:400]: @AleSim94 🧑‍⚕️

**Pull request merged**